extensions/mochi-gg/src/hooks/useWatchList.tsx

```diff
@@ -0,0 +1,122 @@
+import { useCallback, useEffect, useMemo } from "react";
+import { Toast, showToast } from "@raycast/api";
+import axios from "axios";
+import useSWR from "swr";
+import _ from "lodash";
+
+import { IWatchList } from "../wl";
+import useDiscord from "./useDiscord";
+import { IDiscordUser, subscribeOnUserChange } from "../service/discord";
+import { ITickerBaseCoin } from "../type/api";
+
+export default function useWatchList() {
+  const { user } = useDiscord();
+  const { isLoading, isValidating, data, mutate } = useSWR<IWatchList>(
```

Comment:
Note that you can use `useFetch` or `useCachedPromise` from `@raycast/utils` instead that fits better with Raycast extensions.